### PR TITLE
Add C# examples to Using NavigationRegions

### PR DESCRIPTION
**Pull request by ShawnHardern**
Add C# examples to Using NavigationRegions

### CODE REVIEW

File: tutorials/navigation/navigation_using_navigationregions.rst [hunk 1]
```diff
@@ -65,6 +85,18 @@ If regions are created with the NavigationServer API directly they need to be as
...
 
+ .. code-tab:: csharp 2D C#
+
+    public partial class MyClass : Node2D
+    {
+        public override void _Ready()
+        {
+            Rid newRegionRid = NavigationServer2D.RegionCreate();
+            Rid defaultMapRid = GetWorld2D().GetNavigationMap();
+            NavigationServer2D.RegionSetMap(newRegionRid,defaultMapRid);
```
Comment (AThousandShips) [inline tutorials/navigation/navigation_using_navigationregions.rst:96]:
```suggestion
            NavigationServer2D.RegionSetMap(newRegionRid, defaultMapRid);
```

File: tutorials/navigation/navigation_using_navigationregions.rst [hunk 2]
```diff
@@ -74,6 +106,18 @@ If regions are created with the NavigationServer API directly they need to be as
...
 
+ .. code-tab:: csharp 3D C#
+
+    public partial class MyClass : Node3D
+    {
+        public override void _Ready()
+        {
+            Rid newRegionRid = NavigationServer3D.RegionCreate();
+            Rid defaultMapRid = GetWorld3D().GetNavigationMap();
+            NavigationServer3D.RegionSetMap(newRegionRid,defaultMapRid);
```
Comment (AThousandShips) [inline tutorials/navigation/navigation_using_navigationregions.rst:117]:
```suggestion
            NavigationServer3D.RegionSetMap(newRegionRid, defaultMapRid);
```

**Review by smix8 — APPROVED** [review]

File: tutorials/navigation/navigation_using_navigationregions.rst [hunk 3]
```diff
@@ -45,12 +45,32 @@ The region RID can then be obtained from NavigationRegion Nodes with ``get_rid()
...
+
  .. code-tab:: gdscript 3D GDScript
 
     extends NavigationRegion3D
 
     var navigationserver_region_rid: RID = get_rid()
 
+ .. code-tab:: csharp 3D C#
+
+    public partial class MyClass : NavigationRegion3D
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationregions.rst:66]:
```suggestion
    public partial class MyNavigationRegion3D : NavigationRegion3D
```

File: tutorials/navigation/navigation_using_navigationregions.rst [hunk 4]
```diff
@@ -45,12 +45,32 @@ The region RID can then be obtained from NavigationRegion Nodes with ``get_rid()
...
+    {
+        public override void _Ready()
+        {
+            Rid navigationserverRegionRid = GetRid();
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationregions.rst:70]:
```suggestion
            Rid navigationServerRegionRid = GetRid();
```

File: tutorials/navigation/navigation_using_navigationregions.rst [hunk 5]
```diff
@@ -65,6 +85,18 @@ If regions are created with the NavigationServer API directly they need to be as
...
         NavigationServer2D.region_set_map(new_region_rid, default_map_rid)
 
+ .. code-tab:: csharp 2D C#
+
+    public partial class MyClass : Node2D
+    {
+        public override void _Ready()
+        {
+            Rid newRegionRid = NavigationServer2D.RegionCreate();
+            Rid defaultMapRid = GetWorld2D().GetNavigationMap();
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationregions.rst:95]: Prefer properties to hidden methods.

```suggestion
            Rid defaultMapRid = GetWorld2D().NavigationMap;
```

File: tutorials/navigation/navigation_using_navigationregions.rst [hunk 6]
```diff
@@ -74,6 +106,18 @@ If regions are created with the NavigationServer API directly they need to be as
...
         NavigationServer3D.region_set_map(new_region_rid, default_map_rid)
 
+ .. code-tab:: csharp 3D C#
+
+    public partial class MyClass : Node3D
+    {
+        public override void _Ready()
+        {
+            Rid newRegionRid = NavigationServer3D.RegionCreate();
+            Rid defaultMapRid = GetWorld3D().GetNavigationMap();
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationregions.rst:116]:
```suggestion
            Rid defaultMapRid = GetWorld3D().NavigationMap;
```

**Review by raulsntos — APPROVED** [review]
Looks good to me, thanks!

**Comment by ShawnHardern** [comment]
> Looks good to me, thanks!

Awesome thanks @AThousandShips @raulsntos @smix8  for taking the time to review, I have squashed the commits 🥳

**Comment by AThousandShips** [comment]
I would suggest not opening any more separate PRs if you have cases but combine all into one remaining PR, these are tiny changes while important and it makes it a lot easier to review and follow up, they don't need separate PRs

**Comment by ShawnHardern** [comment]
> I would suggest not opening any more separate PRs if you have cases but combine all into one remaining PR, these are tiny changes while important and it makes it a lot easier to review and follow up, they don't need separate PRs

No problem, I'll consolidate any further changes into a single PR. I was aiming to keep the changes small and focused, but totally understand making the review process smoother 😎🥳. Thanks for the feedback!

**Review by AThousandShips — APPROVED** [review]

**Comment by AThousandShips** [comment]
Thanks!

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.3 in https://github.com/godotengine/godot-docs/pull/10038.

**Pull request merged**